src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -501,3 +511,96 @@ export class HttpConnection implements IConnection {
 function transportMatches(requestedTransport: HttpTransportType | undefined, actualTransport: HttpTransportType) {
     return !requestedTransport || ((actualTransport & requestedTransport) !== 0);
 }
+
+export class TransportSendQueue {
+    private buffer: any[] = [];
+    private sendBufferedData: PromiseSource;
+    private executing: boolean = true;
+    private transportResult: PromiseSource;
+
+    constructor(private transport: ITransport) {
```

Comment:
Does this make a member variable on the class and automatically assign it?

Not sure how I feel about it...